src/takeLastWhile.ts

```diff
@@ -0,0 +1,46 @@
+import { purry } from './purry';
+
+/**
+ * Returns elements from the end of the array until the predicate returns false.
+ *
+ * @param array the array
+ * @param fn the predicate
+ * @signature
+ *    R.takeLastWhile(array, fn)
+ * @example
+ *    R.takeLastWhile([1, 2, 3, 4, 3, 2, 1], x => x !== 4) // => [3, 2, 1]
+ * @dataFirst
+ * @category Array
+ */
+export function takeLastWhile<T>(
+  array: ReadonlyArray<T>,
+  fn: (item: T) => boolean
+): Array<T>;
+
+/**
+ * Returns elements from the end of the array until the predicate returns false.
+ *
+ * @param fn the predicate
+ * @signature
+ *    R.takeLastWhile(fn)(array)
+ * @example
+ *    R.pipe([1, 2, 3, 4, 3, 2, 1], R.takeLastWhile(x => x !== 4))  // => [3, 2, 1]
+ * @dataLast
+ * @category Array
+ */
+export function takeLastWhile<T>(
+  fn: (item: T) => boolean
+): (array: ReadonlyArray<T>) => Array<T>;
+
+export function takeLastWhile() {
+  return purry(_takeLastWhile, arguments);
+}
+
+function _takeLastWhile<T>(array: Array<T>, fn: (item: T) => boolean) {
```

Comment:
I think the implementation could support lazy evaluation, but `takeWhile` doesn't support it either, so maybe that should be a separate task to keep these functions consistent.

Reply:
It can't; when running lazily we go over the items from start to end and need to make a decision based on the items we already saw. Here the first item only knows if it should be returned or dropped once a further item comes along and returns a false value for the predicate.

takeWhile can be lazy because items have all the info they need while processing.

Reply:
Ok. Interesting... I would imagine it's possible to do the same thing from end to start. But nevermind, I can look at the implementation if I ever feel like I want to understand how the laziness really works. :)